internal/xdscache/v3/listener.go

```diff
@@ -425,12 +422,25 @@ func (c *ListenerCache) OnChange(root *dag.DAG) {
 					DelayedCloseTimeout(cfg.Timeouts.DelayedClose).
 					MaxConnectionDuration(cfg.Timeouts.MaxConnectionDuration).
 					ConnectionShutdownGracePeriod(cfg.Timeouts.ConnectionShutdownGracePeriod).
-					AllowChunkedLength(cfg.AllowChunkedLength).
-					AddFilter(envoy_v3.OriginalIPDetectionFilter(cfg.XffNumTrustedHops)).
-					AddFilter(envoy_v3.GlobalRateLimitFilter(envoyGlobalRateLimitConfig(cfg.RateLimitConfig))).
-					Get()
+					AllowChunkedLength(cfg.AllowChunkedLength)
+
+				if vh.AuthorizationService != nil && vh.AuthBeforeRateLimiting {
+					hcmBuilder.
+						AddFilter(envoy_v3.FilterMisdirectedRequests(vh.VirtualHost.Name)).
+						DefaultFilters().
+						AddFilterBefore(authFilter, "local_ratelimit").
```

Comment:
This `AddFilterBefore` function is a little ugly, but pulling the local rate limit filter out of `DefaultFilters()` would've been painful. Open to any other ideas.

Reply:
yeah only other thing i can think of besides pulling out that filter is a bigger refactor, so that the order you call `AddFilter` doesn't matter, we maybe always have a categorized sort order for filters (that can be slightly switched based on `AuthBeforeRateLimiting` ofc), but that could be messy and inflexible later